src/include/sof/math/fir_config.h

```diff
@@ -21,24 +21,26 @@
 #endif
 #endif
 
-/* If next defines are set to 1 the EQ is configured automatically. Setting
- * to zero temporarily is useful is for testing needs.
- * Setting EQ_FIR_AUTOARCH to 0 allows to manually set the code variant.
+/* Define SOFM_FIR_FORCEARCH yes/2/3 in build command line to override default build
+ * for gcc and xcc compilers. Leave it undefined normally.
  */
-#define FIR_AUTOARCH    1
-
-/* Force manually some code variant when EQ_FIR_AUTODSP is set to zero. These
- * are useful in code debugging.
- */
-#if FIR_AUTOARCH == 0
+#ifdef SOFM_FIR_FORCEARCH
+#if SOFM_FIR_FORCEARCH == 3
 #define FIR_GENERIC	0
 #define FIR_HIFIEP	0
 #define FIR_HIFI3	1
+#elif SOFM_FIR_FORCEARCH == 2
+#define FIR_GENERIC	0
+#define FIR_HIFIEP	1
+#define FIR_HIFI3	0
+#else
+#define FIR_GENERIC	1
+#define FIR_HIFIEP	0
+#define FIR_HIFI3	0
```

Comment:
@singalsu @andrula-song can we make all these Kconfig options for v2.8. Thanks !

Reply:
I dont mean a system wide HIFI level, but a per module Kconfig menu where we can set generic | hifi2 | hifi3 | avx etc